### PR TITLE
libdnf: switch back to default stdenv.

### PR DESCRIPTION
**Pull request by rb2k**
###### Motivation for this change

The only reason this was on GCC was because of some implicit constructor conversion issues that gcc handled without complaining and clang did not.

This commit fixed those:
https://github.com/rpm-software-management/libdnf/commit/fefe0b69e8368a8e9ceeed107add174c64f00f5e

Now that it works on clang, might as well revert back to the default.

Note: I can't currently test this on darwin because libbfd in master seem broken even without my modifications. I just end up with this: https://www.internalfb.com/phabricator/paste/view/P459437194

I did however apply the fefe0b69e8368a8e9ceeed107add174c64f00f5e commit as a patch to 21.05 and was able to also compile it with stdenv.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alyssais** [comment]
This commit message is misleading — what this change does is switch back to the default stdenv, which on most platforms _is_ the gcc stdenv, not the clang one.  Could you rephrase it?

**Comment by rb2k** [comment]
> This commit message is misleading — what this change does is switch back to the default stdenv, which on most platforms _is_ the gcc stdenv, not the clang one. Could you rephrase it?

Yeah, you're right. That's better. PR is still in draft form and not quite ready, so I'll add some more details over the next few minutes :)

**Comment by rb2k** [comment]
Result of `nixpkgs-review pr 138540` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>libdnf</li>
    <li>microdnf</li>
  </ul>
</details>

**Comment by alyssais** [comment]
The commit message still doesn't make sense — the version you changed the PR title to is good, can you use that as the commit message too?

**Comment by rb2k** [comment]
> The commit message still doesn't make sense — the version you changed the PR title to is good, can you use that as the commit message too?

ugh, sorry. clearly I need more coffee this morning :)
done

**Comment by alyssais** [comment]
@ofborg build libdnf.override { stdenv = clangStdenv; }

**Comment by alyssais** [comment]
Hmm, that appeared not to work.  Can I get it to work like this?

@OfBorg build libdnf.override{stdenv=clangStdenv;}

**Comment by rb2k** [comment]
Found the issue with darwin on master: https://github.com/NixOS/nixpkgs/pull/138545 (indeed unrelated to this)

**Review by alyssais — APPROVED** [review]
Tested on x86_64-linux with

    nix-build -E 'with import <nixpkgs> {}; libdnf.override { stdenv = clangStdenv; }' -I nixpkgs=https://github.com/NixOS/nixpkgs/tarball/pull/138540/head
